lib/elixir/lib/option_parser.ex

```diff
@@ -543,6 +545,10 @@ defmodule OptionParser do
           {strict, true}
 
         true ->
+          IO.warn(
+            "Dynamic mode in OptionParser is deprecated, please consider using `strict: [...]`"
```

Comment:
I think a better warning would be:

    "Not passing the :switches or :strict option to OptionParser is deprecated".